### PR TITLE
Search task added

### PR DESCRIPTION
**Pull request by Nurlanprog**
Function accepts json string or in dict format. Test changes are added to check whether function will accept string and dict as input request body parameters.
 
### Issues Resolved
[[List any issues this PR will resolve]](https://github.com/opensearch-project/opensearch-py-ml/issues/104)
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by Nurlanprog** [comment]
Code of search task and search model will be the same. Only url differs. Maybe should I add boolean parameter to function to check whether user searching model or task. In example: search(model=True, json="...") or search(task=True, json="...").

Or is it inconvenient to customers?

**Comment by dhrubo-os** [comment]
> 

Good point. But for simplification, may be just keep two separate function. We can revise later if we need to.

**Comment by Nurlanprog** [comment]
Ok

File: opensearch_py_ml/ml_commons/ml_commons_client.py [hunk 1]
```diff
@@ -359,6 +360,34 @@ def _get_task_info(self, task_id: str):
...
+        if isinstance(json_input, str):
+            try:
+                json_obj = json.loads(json_input)
+                API_BODY = json.dumps(json_obj)
+            except json.JSONDecodeError:
+                return "Invalid JSON string passed as argument."
+        elif isinstance(json_input, dict):
+            API_BODY = json.dumps(json_input)
+        else:
+            return "Invalid JSON object passed as argument."
```
Comment (dhrubo-os) [inline opensearch_py_ml/ml_commons/ml_commons_client.py:383]: Can we cover this line with testing?

Comment (dhrubo-os) [reply opensearch_py_ml/ml_commons/ml_commons_client.py:383]: "return "Invalid JSON object passed as argument." this block

Comment (Nurlanprog) [reply opensearch_py_ml/ml_commons/ml_commons_client.py:383]: Ah I see maybe you mean try catch block missed for dict case. I will add it.

Comment (Nurlanprog) [reply opensearch_py_ml/ml_commons/ml_commons_client.py:383]: > "return "Invalid JSON object passed as argument." this block

Ok

File: tests/ml_commons/test_ml_commons_client.py [hunk 2]
```diff
@@ -372,6 +372,26 @@ def test_integration_model_train_register_full_cycle():
                     print("Model Task Status:", ml_task_status)
                     raised = True
                 assert raised == False, "Raised Exception in pulling task info"
+
+                try:
+                    search_task_obj = ml_client.search_task(json="{'_id':'{task_id}'}")
```
Comment (dhrubo-os) [inline tests/ml_commons/test_ml_commons_client.py:377]: I saw parameter name was: `json_input`. Then how can we invoke with `json`?

File: tests/ml_commons/test_ml_commons_client.py [hunk 3]
```diff
@@ -372,6 +372,26 @@ def test_integration_model_train_register_full_cycle():
...
+                except:  # noqa: E722
+                    raised = True
+                assert raised == False, "Raised Exception in searching task"
+
+                try:
+                    search_task_obj = ml_client.search_task(json=15)
+                    assert search_task_obj == "Invalid JSON object passed as argument."
+                except:  # noqa: E722
+                    raised = True
+                assert raised == False, "Raised Exception in searching task"
```
Comment (dhrubo-os) [inline tests/ml_commons/test_ml_commons_client.py:394]: can we make a separate function for this to test? This chain of flow test was to test integration test. We don't need to include this test here.

File: tests/ml_commons/test_ml_commons_client.py (hunk 2 above)
Comment (Nurlanprog) [reply tests/ml_commons/test_ml_commons_client.py:377]: Yes, my bad sorry. I will change it.

File: tests/ml_commons/test_ml_commons_client.py (hunk 3 above)
Comment (Nurlanprog) [reply tests/ml_commons/test_ml_commons_client.py:394]: > can we make a separate function for this to test? This chain of flow test was to test integration test. We don't need to include this test here.

Ok.

File: tests/common.py [hunk 4]
```diff
@@ -57,7 +57,7 @@
 )
 _pd_ecommerce.insert(2, "customer_birth_date", None)
 _pd_ecommerce.index = _pd_ecommerce.index.map(str)  # make index 'object' not int
-_pd_ecommerce["customer_birth_date"].astype("datetime64")
+_pd_ecommerce["customer_birth_date"].astype("datetime64[ns]")
```
Comment (dhrubo-os) [inline tests/common.py:60]: Why did we need to change this?

Comment (Nurlanprog) [reply tests/common.py:60]: Are you not encountered with this error?

**Review by dhrubo-os — APPROVED** [review]

**Review by dhrubo-os — APPROVED** [review]

File: tests/ml_commons/test_ml_commons_client.py [hunk 5]
```diff
@@ -100,6 +100,102 @@ def test_execute():
     ), "Raised Exception during execute API testing with JSON string"
 
 
+def test_search():
```
Comment (b4sjoo) [inline tests/ml_commons/test_ml_commons_client.py:103]: Will it be better if we print out error message?

**Review by rbhavna — APPROVED** [review]

**Pull request merged**